OpenRA.Mods.Common/Infiltration/InfiltrateForPlayerUpgrades.cs

```diff
@@ -0,0 +1,45 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Grants upgrades to the infiltrating actor's owner.")]
+	public class InfiltrateForPlayerUpgradesInfo : ITraitInfo
+	{
+		[Desc("Upgrades to grant when this actor is infiltrated."), FieldLoader.Require]
```

Comment:
This is missing `[UpgradeGrantedReference]`. See https://github.com/OpenRA/OpenRA/pull/8321.